engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java

```diff
@@ -86,21 +94,28 @@ private long handleInterrupting(
   private boolean interruptParentScope(final BpmnStepContext<T> context) {
     final long scopeKey = context.getValue().getFlowScopeKey();
     final List<ElementInstance> children = context.getElementInstanceState().getChildren(scopeKey);
-    boolean waitForTermination = false;
+
+    int terminatedChildInstances = 0;
 
     for (final ElementInstance child : children) {
       if (child.canTerminate()) {
-        waitForTermination = true;
         context
             .getOutput()
             .appendFollowUpEvent(
                 child.getKey(), WorkflowInstanceIntent.ELEMENT_TERMINATING, child.getValue());
-      } else {
-        context.getElementInstanceState().consumeToken(scopeKey);
+
+        terminatedChildInstances += 1;
       }
     }
 
-    return waitForTermination;
+    // consume all other active tokens (e.g. tokens waiting at a joining gateway)
+    final int zombies =
+        context.getFlowScopeInstance().getNumberOfActiveTokens() - terminatedChildInstances;
+    for (int z = 0; z < zombies; z++) {
```

Comment:
Just a minor nit pick: This was a little odd to me, I guess I'm so used to seeing a `for` when iterating, that I had to read the next line again looking for the `z` variable. It also seems that we can achieve the same without it, like:

```suggestion
    while (zombies-- > 0) {
```
or
```suggestion
while (zombies > 0) {
zombies--;
...
```

Reply:
Thanks @MiguelPires. I see your point. But as for now, I will not change it. There are 2 other places which almost the same code :see_no_evil: I hope to get rid of it soon.